### ISSUE TITLE
CH 31 Form Payload Redirect

### ISSUE THREAD

**Issue by steele-lm** [department-of-veterans-affairs/va.gov-team#65386]
## Issue Description
_The Chapter 31 digital form payload is currently sent to CMS, which is being retired. The payload needs to be redirected to point to the RES system._

---
## Tasks
- [ ] Change the Chapter 31 json endpoint from CMS to the new RES endpoint. Here is the link to the [documentation in SharePoint](https://dvagov.sharepoint.com/:w:/r/sites/RESIntegrations594/Shared%20Documents/VA.gov/RES%20%E2%80%98Push%20Form%2028-1900%20Application%20Data%E2%80%99%20Service%20Contract.docx?d=wb8e0279bca754d3b9c16112856a8e421&csf=1&web=1&e=f0VLBb).

## Acceptance Criteria
- [ ] The Chapter 31 json payload is received by the RES system.
---

**Comment by steele-lm** [comment]
Confirm if we should send json file to both CMS and RES as a fail-safe.

**Comment by steele-lm** [comment]
Ekaterina (Katya) Shick confirmed that the payload no longer needs to be sent to CMS once it's redirected to RES.

**Comment by TaiWilkin** [comment]
Progress update: We're currently working on adding this as an external service to the forward proxy, which will allow us to test the code changes locally and update the CI tests. Work is progressing slowly due to a lack of access to required systems.

**Comment by TaiWilkin** [comment]
After some investigation, we've determined that before we can successfully connect to the RES endpoint from within the vets-api, we need to add it as a new External Service Integration within each environment (not just for testing, as we initially thought), allowing access via the fwdproxy instance. This is progressing slowly as we need to request various access permissions in order to complete the work, but once the appropriate access is granted we should be able to move more quickly. 

The update will also require coordination across multiple repositories, so we've also evaluated options for safely rolling this out without a disruption to current functionality. The planned approach is to add the RES endpoint as a new and separate endpoint within all appropriate repositories, while maintaining the existing VRE CMS endpoint. This will enable us to use a flipper for the API update and keep sending data to the VRE CMS in environments with the flipper turned off until the endpoint has been added to and tested within each environment. 

As of now, the following steps for implementation are planned:

1. [Add RES endpoint to dev fwdproxy as new external service](https://github.com/department-of-veterans-affairs/devops/pull/13604) - **completed**
2. Add RES API key to [AWS param store in dev](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-on-eks#VetsAPIonEKS-SecretValues) - blocked while waiting for AWS access; [access request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68857)
3. [Add a flipper](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide) to support the endpoint changeover in the backend (so that the change isn’t propagated through all environments before the external service exists)
4. [Update the API to send the data to the RES endpoint](https://github.com/department-of-veterans-affairs/vets-api/pull/14179) (with additional requested fields) - draft PR on hold while waiting on steps 1-3; will require additional updates prior to testing to reflect changed approach
5. Turn on flipper in development and test the new submission process with VRE to confirm that they receive the data
6. Add RES endpoint and API key to sandbox/QA, then turn on the flipper in sandbox/QA; test with VRE to confirm that they receive the data
7. Add RES endpoint and API key to staging, then turn on the flipper in staging; test with VRE to confirm that they receive the data - **blocked** pending creation of staging endpoint
8. Add RES endpoint and API key to production, then turn on the flipper in production; test with VRE to confirm that they receive the data - **blocked** pending creation of production endpoint
9. Remove flipper and old code from API
10. Remove VRE CMS endpoint and VRE CMS secrets in all environments

**Notes**: 
This plan of implementation is based on my current understanding of the process of adding external services, which is based on a combination of documentation and code analysis. Some of my initial assumptions have already been proven incorrect, and some documentation has been incorrect/out-of-date, so it's possible there are additional steps that I'm not yet aware of. 
In addition, there will likely be some monitoring and documentation steps related to the addition of the new endpoint, but I believe these can be pursued on a separate work path.

**Comment by TaiWilkin** [comment]
Handing this off to the non-disability team due to the extent of work remaining.
The current [draft PR can be found here](https://github.com/department-of-veterans-affairs/vets-api/pull/14179). Next steps for it are to add the API key, add logs for testing purposes, and use the build preview + logs to determine whether the forwarding for the endpoint is working correctly.

**Comment by aplatt-coforma** [comment]
@vbellissimo - should we take the "pension" label off of this ticket since our team is no longer working on it? 

cc @steele-lm

**Comment by sortizsh** [comment]
Handoff to Micah

**Comment by vbellissimo** [comment]
@aplatt-coforma , yes I am okay if we take the Pension label off this.

**Comment by sortizsh** [comment]
@SloopKoning  can you please look into this?

**Comment by MadBomber** [comment]
I have identified the places within the `vets-api` codebase where some changes may need to be made.  What those changes are is still TBD.   `lib/vre` is the primary place to look.  search terms to use through the coe base are "ch31" and "veteran_readiness"

**Comment by sortizsh** [comment]
Laura sent Dewayne some documents to review today.

**Comment by steele-lm** [comment]
@MadBomber Here is the RES documentation that was on Sharepoint: [RES ‘Push Form 28-1900 Application Data’ Service Contract.docx](https://github.com/department-of-veterans-affairs/va.gov-team/files/14451860/RES.Push.Form.28-1900.Application.Data.Service.Contract.docx)

**Comment by sortizsh** [comment]
@SloopKoning

**Comment by sortizsh** [comment]
we need to provide a timeline to Laura

**Comment by sortizsh** [comment]
Laura met with RES team and provided the timeline. end of April

**Comment by steele-lm** [comment]
RES team confirmed we can push to prod when ready.
